### DISCUSSION TITLE
start execution via API

### DISCUSSION THREAD

**Discussion by ncouronne** [flyteorg/flyte#1250] [Q&A] [answered]
I'm looking for launching execution via a API rather than with the flytecli.
Do you have any example to do so, i can't find  in the documentation ?

Do you have advices to use REST and gRPC interface to flyte admin service ?

**Comment by kumare3** [comment, marked as answer]
Hi @ncouronne we absolutely advice to use the gRpc / REST interface of FlyteAdmin Service. But, we advice using through one of the supported clients at the moment
1. [flytekit raw client](https://docs.flyte.org/projects/flytekit/en/latest/design/clis.html#clients)
2. [Official Golang Client](https://github.com/flyteorg/flyteidl/blob/master/clients/go/admin/client.go). Example usage - [flytectl](https://github.com/flyteorg/flytectl/blob/5969b182fcb319c1b596dc14b2aa7b3f2dd79cdf/cmd/core/cmd.go#L61)

Flytectl contains some extensions to the base client - [here](https://github.com/flyteorg/flytectl/tree/master/pkg/ext) - we might move them to flyteidl to make it available for everyone!

Also coming soon is a simplified [flytekit pythonic remote interface for Flyte backend](https://github.com/flyteorg/flytekit/tree/master/flytekit/remote). We want to release this as part of 0.16.0 (early august) - This would be the most recommended and natural way to integrate
